src/Backend/Modules/Faq/Actions/DeleteCategory.php

```diff
@@ -24,37 +24,48 @@ public function execute(): void
         $deleteForm = $this->createForm(FaqCategoryDeleteType::class);
         $deleteForm->handleRequest($this->getRequest());
         if (!$deleteForm->isSubmitted() || !$deleteForm->isValid()) {
-            $this->redirect(BackendModel::createURLForAction('Categories') . '&error=something-went-wrong');
+            $this->redirect(BackendModel::createURLForAction(
+                'Categories',
+                null,
+                null,
+                ['error' => 'something-went-wrong']
+            ));
+
+            return;
         }
         $deleteFormData = $deleteForm->getData();
 
         $this->id = $deleteFormData['id'];
 
         // does the item exist
-        if ($this->id !== 0 && BackendFaqModel::existsCategory($this->id)) {
-            $this->record = (array) BackendFaqModel::getCategory($this->id);
-
-            if (BackendFaqModel::deleteCategoryAllowed($this->id)) {
-                parent::execute();
-
-                // delete item
-                BackendFaqModel::deleteCategory($this->id);
-
-                // category was deleted, so redirect
-                $this->redirect(
-                    BackendModel::createURLForAction('Categories') . '&report=deleted-category&var=' .
-                    rawurlencode($this->record['title'])
-                );
-            } else {
-                $this->redirect(
-                    BackendModel::createURLForAction('Categories') . '&error=delete-category-not-allowed&var=' .
-                    rawurlencode($this->record['title'])
-                );
-            }
-        } else {
-            $this->redirect(
-                BackendModel::createURLForAction('Categories') . '&error=non-existing'
-            );
+        if ($this->id === 0 || !BackendFaqModel::existsCategory($this->id)) {
+            $this->redirect(BackendModel::createURLForAction('Categories', null, null, ['error' => 'non-existing']));
+
+            return;
         }
+
+        $this->record = (array) BackendFaqModel::getCategory($this->id);
+
+        if (!BackendFaqModel::deleteCategoryAllowed($this->id)) {
+            $this->redirect(BackendModel::createURLForAction(
+                'Categories',
+                null,
+                null,
+                ['error' => 'delete-category-not-allowed', 'var' => $this->record['title']]
```

Comment:
Hmm, haven't looked at that yet, still working on the general delete form type. Do I change the tests to +? Or do I change the http_query_build enc_type to PHP_QUERY_RFC3986 so spaces are translated back to %20?